lib/active_resource/base.rb

```diff
@@ -1239,7 +1284,9 @@ def initialize(attributes = {}, persisted = false)
     #   not_ryan.hash            # => {:not => "an ARes instance"}
     def clone
       # Clone all attributes except the pk and any nested ARes
-      cloned = attributes.reject { |k, v| k == self.class.primary_key || v.is_a?(ActiveResource::Base) }.transform_values { |v| v.clone }
+      # TODO could @attributes.clone suffice?
+      cloned = ActiveModel::AttributeSet.new({})
+      @attributes.each_value { |v| cloned[v.name] = v.clone if !(v.name == self.class.primary_key.to_s || v.value.is_a?(ActiveResource::Base)) }
```

Comment:
What is the history behind excluding primary keys and `ActiveResource::Base` instances?

Would calling `ActiveModel::AttributeSet#clone` (through `@attributes.clone`) be sufficient?

If not, what is another way to efficiently clone the attributes in a way that meets the requirements?